plugin/pkg/auth/authorizer/node/node_authorizer.go

```diff
@@ -0,0 +1,169 @@
+//    node -> pod
+//    node -> pod -> secret
+//    node -> pod -> configmap
+//    node -> pod -> pvc
+//    node -> pod -> pvc -> pv
+//    node -> pod -> pvc -> pv -> secret
+type NodeAuthorizer struct {
+	graph      *Graph
+	identifier nodeidentifier.NodeIdentifier
+	strict     bool
+	nodeRules  []rbacapi.PolicyRule
+}
+
+// New returns a new node authorizer
+func New(graph *Graph, identifier nodeidentifier.NodeIdentifier, strict bool) authorizer.Authorizer {
+	return &NodeAuthorizer{
+		graph:      graph,
+		identifier: identifier,
+		strict:     strict,
+		nodeRules:  bootstrappolicy.NodeRules(),
+	}
+}
+
+func (r *NodeAuthorizer) Authorize(attrs authorizer.Attributes) (bool, string, error) {
+	nodeName, isNode := r.identifier.NodeIdentity(attrs.GetUser())
+	if !isNode {
+		// reject requests from non-nodes
+		return false, "", nil
+	}
+	if len(nodeName) == 0 && r.strict {
+		// reject requests from unidentifiable nodes if in strict mode
+		return false, "unknown node", nil
+	}
+
+	if !rbac.RulesAllow(attrs, r.nodeRules...) {
+		// reject requests that don't fit within the broad authorization rules
+		return false, "", nil
+	}
+
+	if len(nodeName) == 0 {
+		// allow requests from unidentifiable nodes if not in strict mode
+		return true, "", nil
+	}
+
+	// subdivide access to specific resources
+
+	if attrs.GetAPIGroup() != api.GroupName {
+		// the resources we care about subdividing happen to be in the core API group.
+		// access to resources in other API groups is not subdivided.
+		return true, "", nil
+	}
+
+	switch attrs.GetResource() {
+	case "configmaps":
+		if attrs.GetVerb() != "get" || len(attrs.GetName()) == 0 || len(attrs.GetNamespace()) == 0 {
+			glog.V(2).Infof("NODE DENY: %#v", attrs)
+			return false, "can only get individual configmaps", nil
+		}
+		return r.hasPathFrom(nodeName, ConfigMapNode, attrs.GetNamespace(), attrs.GetName()), "", nil
+
+	case "secrets":
+		if attrs.GetVerb() != "get" || len(attrs.GetName()) == 0 || len(attrs.GetNamespace()) == 0 {
+			glog.V(2).Infof("NODE DENY: %#v", attrs)
+			return false, "can only get individual secrets", nil
+		}
+		return r.hasPathFrom(nodeName, SecretNode, attrs.GetNamespace(), attrs.GetName()), "", nil
+
+	case "persistentvolumeclaims":
+		if attrs.GetVerb() != "get" || len(attrs.GetName()) == 0 || len(attrs.GetNamespace()) == 0 {
+			glog.V(2).Infof("NODE DENY: %#v", attrs)
+			return false, "can only get individual persistentvolumeclaims", nil
+		}
+		return r.hasPathFrom(nodeName, PVCNode, attrs.GetNamespace(), attrs.GetName()), "", nil
+
+	case "persistentvolumes":
+		if attrs.GetVerb() != "get" || len(attrs.GetName()) == 0 {
+			glog.V(2).Infof("NODE DENY: %#v", attrs)
+			return false, "can only get individual persistentvolumes", nil
+		}
+		return r.hasPathFrom(nodeName, PVNode, "", attrs.GetName()), "", nil
+
```

Comment:
not sure I understand this comment. I actually moved the system:node rule check to the end, so now the function looks like:

1. reject requests that aren't from nodes
2. specially handle requests for secret/configmap/pvc/pv objects
3. for other resources, defer to the system:node rule evaluation